dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/info/OpenTelemtryLifecycleManagerImpl.java

```diff
@@ -59,39 +59,29 @@ public OpenTelemtryLifecycleManagerImpl(@Reference MetaDataSlotService slotServi
 
         String shortName = (String) featureRef.getProperty("ibm.featureName");
         //the runtimeMode was added in 2.0
-        final boolean telemetry2OrLater = shortName.startsWith("mpTelemetry-1") ? false : true;
+        telemetry2OrLater = shortName.startsWith("mpTelemetry-1") ? false : true;
 
-        CheckpointHook checkpointHook = new CheckpointHook() {
-
-            @Override
-            public void restore() {
-                waitingForCheckpointRestore = false;
-                checkThenSetRuntimeFields(telemetry2OrLater);
-            }
-        };
-
-        // If checkpoint is enabled this statement will register the hook we created above. Then checkpoint returns true so the if statement is false.
-        // If checkpoint is disabled it does nothing and returns false, so the if statement will be true.
-        if (!CheckpointPhase.getPhase().addMultiThreadedHook(checkpointHook)) {
-            //Checkpoint disabled
-            waitingForCheckpointRestore = false;
-            checkThenSetRuntimeFields(telemetry2OrLater);
-        } else {
-            //Checkpoint enabled, before restore.
-            waitingForCheckpointRestore = true;
-        }
+        checkThenSetRuntimeFields();
     }
 
-    private void checkThenSetRuntimeFields(boolean telemetry2OrLater) {
-        HashMap<String, String> propreties = OpenTelemetryPropertiesReader.getRuntimeInstanceTelemetryProperties();
-        isRuntimeEnabled = telemetry2OrLater && !!!OpenTelemetryPropertiesReader.checkDisabled(propreties);
-
-        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
-            Tr.debug(this, tc, "Configured runtime mode as ", isRuntimeEnabled);
-        }
-
-        if (isRuntimeEnabled) {
-            runtimeInstance = LazyInitializer.<OpenTelemetryInfo> builder().setInitializer(curryInfoFactory(isRuntimeEnabled)).get();
+    private void checkThenSetRuntimeFields() {
+        if (waitingForCheckpointRestore) {
+            synchronized (this) {
+                //restored() true if checkpoint has been restored or was never active
+                if (waitingForCheckpointRestore && CheckpointPhase.getPhase().restored()) {
+                    waitingForCheckpointRestore = false;
```

Comment:
Should set `waitingForCheckpointRestore` as the last thing since it is also checked before entering the synchronization block and we need to ensure the `isRuntimeEnabled` and `runtimeInstance` are set appropriately before reporting `waitingForCheckpointRestore` is false.